### PR TITLE
Archive choices for interfaces

### PR DESCRIPTION
**Pull request by dylant-da**
Automatically generate `HasArchive` and `HasExercise Archive` instances for all interfaces

Closes https://github.com/digital-asset/daml/issues/12556
https://github.com/digital-asset/ghc/pull/135

### CODE REVIEW

**Comment by remyhaemmerle-da** [comment]
Could you extend a bit the description and link the ghc branch ?

**Comment by remyhaemmerle-da** [comment]
I like the compiler output I can see, though ;)

**Comment by akrmn** [comment]
I held off on this given @cocreature's https://github.com/digital-asset/daml/issues/12556#issuecomment-1239270360,

> I discussed making Archive a "builtin" choice with @remyhaemmerle-da and I’d do it as part of that rather than just generating it from the compiler.

I don't think there's any problem if we do that later, though, since at the language level it's still a choice named `Archive`.

> Could you extend a bit the description and link the ghc branch ?

+1

File: bazel_tools/ghc-lib/version.bzl [hunk 1]
```diff
@@ -9,7 +9,7 @@ GHC_LIB_PATCHES = [
 ]
 
 GHC_REPO_URL = "https://github.com/digital-asset/ghc"
-GHC_REV = "c119f3bdad49336c2b8d17e04d7070ef419132b8"
+GHC_REV = "bbe22fa527a6db95534c1fe70163bcfff8a88eb6"
```
Comment (akrmn) [inline bazel_tools/ghc-lib/version.bzl:12]: please remember to update after merging https://github.com/digital-asset/ghc/pull/135

**Review by akrmn — APPROVED** [review]
Thank you!

**Comment by akrmn** [comment]
Another thing @dylant-da, please add a changelog entry mentioning that interfaces will gain the `Archive` choice and the `archive` method

**Pull request merged**